tracer/src/Datadog.Tracer.Native/iast/iast_util.cpp

```diff
@@ -0,0 +1,1265 @@
+        size_t off = 0;
+        if (offset)
+        {
+            off = *offset;
+        }
+        auto pos = where.find(what, off);
+        if (pos != WSTRING::npos && offset)
+        {
+            *offset = pos + what.length();
+        }
+        return pos;
+    }
+    std::size_t IndexOf(const std::string& where, const std::string& what, std::size_t* offset)
+    {
+        size_t off = 0;
+        if (offset)
+        {
+            off = *offset;
+        }
+        auto pos = where.find(what, off);
+        if (pos != std::string::npos && offset)
+        {
+            *offset = pos + what.length();
+        }
+        return pos;
+    }
+
+    std::size_t LastIndexOf(const WSTRING& where, const WSTRING& what, std::size_t* offset)
+    {
+        size_t off = -1;
+        if (offset)
+        {
+            off = *offset;
+        }
+        auto pos = where.find_last_of(what, off);
+        if (pos != WSTRING::npos && offset)
+        {
+            *offset = pos + what.length();
+        }
+        return pos;
+    }
+    std::size_t LastIndexOf(const std::string& where, const std::string& what, std::size_t* offset)
+    {
+        size_t off = -1;
+        if (offset)
+        {
+            off = *offset;
+        }
+        auto pos = where.find_last_of(what, off);
+        if (pos != std::string::npos && offset)
+        {
+            *offset = pos + what.length();
+        }
+        return pos;
+    }
+
+    bool Contains(const WSTRING& where, const WSTRING& what)
+    {
+        return IndexOf(where, what) != WSTRING::npos;
+    }
+    bool Contains(const std::string& where, const std::string& what)
+    {
+        return IndexOf(where, what) != std::string::npos;
+    }
+
+    typedef unsigned char uchar;
+    static const std::string b = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/"; //=
+    std::string ToBase64(const std::string& in)
+    {
+        std::string out;
+
+        int val = 0, valb = -6;
+        for (uchar c : in)
+        {
+            val = (val << 8) + c;
+            valb += 8;
+            while (valb >= 0)
+            {
+                out.push_back(b[(val >> valb) & 0x3F]);
+                valb -= 6;
+            }
+        }
+        if (valb > -6) out.push_back(b[((val << 8) >> (valb + 8)) & 0x3F]);
+        while (out.size() % 4) out.push_back('=');
+        return out;
+    }
+
+    std::string FromBase64(const std::string& in)
+    {
+        std::string out;
+
+        std::vector<int> T(256, -1);
+        for (int i = 0; i < 64; i++) T[b[i]] = i;
+
+        int val = 0, valb = -8;
+        for (uchar c : in)
+        {
+            if (T[c] == -1) break;
+            val = (val << 6) + T[c];
+            valb += 6;
+            if (valb >= 0)
+            {
+                out.push_back(char((val >> valb) & 0xFF));
+                valb -= 8;
+            }
+        }
+        return out;
+    }
+
+    HRESULT GuidFromString(const std::string& str, GUID* pUid)
```

Comment:
you already have `make_guid` functions in here https://github.com/DataDog/dd-trace-dotnet/blob/master/shared/src/native-src/string.h#L51